BungeeCord-Patches/0060-Additional-DoS-mitigations.patch

```diff
@@ -163,10 +168,42 @@ index 388f6cdb..53575ce0 100644
 +    public int expectedMinLength(ByteBuf buf, ProtocolConstants.Direction direction, int protocolVersion) {
 +        return expectedMaxLength(buf, direction, protocolVersion);
 +    }
++    // Waterfall end
+ }
+diff --git a/protocol/src/main/java/net/md_5/bungee/protocol/packet/Handshake.java b/protocol/src/main/java/net/md_5/bungee/protocol/packet/Handshake.java
+index 70934a5b..b0a25b59 100644
+--- a/protocol/src/main/java/net/md_5/bungee/protocol/packet/Handshake.java
++++ b/protocol/src/main/java/net/md_5/bungee/protocol/packet/Handshake.java
+@@ -7,6 +7,7 @@ import lombok.EqualsAndHashCode;
+ import lombok.NoArgsConstructor;
+ import net.md_5.bungee.protocol.AbstractPacketHandler;
+ import net.md_5.bungee.protocol.DefinedPacket;
++import net.md_5.bungee.protocol.ProtocolConstants;
+ 
+ @Data
+ @NoArgsConstructor
+@@ -43,4 +44,19 @@ public class Handshake extends DefinedPacket
+     {
+         handler.handle( this );
+     }
++
++    // Waterfall start: Additional DoS mitigations
++    // newer protocol limits host string to 255 chars (also reasonable for older clients) + allow \0FML\0 afterwards
++    public static final int EXPECTED_MAX_LENGTH = 5 + ((255 * 4 + 3) + 5) + 2 + 1;
```

Comment:
This will have to be increased or removed - Minecraft-specific DDoS protection services such as TCPShield and Cosmic Guard like to stuff extra data in the handshake address field.

Reply:
I will put in a property switch to set the maximum size for that packet then.

I am sure there are many waterfall users not using such ddos protections, they need the limit.

Reply:
255 characters is plenty for our (TCPShield) purposes.

Reply:
It is also multiplied by 4 for the heaviest utf-8 chars taking 4 bytes, so its actually much more, if you just use 1 or 2 byte utf-8 chars only.
Then I guess it is not needed in this case to allow editing this value.